# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1291685447 28800
#      Mon Dec 06 17:30:47 2010 -0800
# Node ID c715816f4539dfad06fb83882d66d9cf3e0ddda0
# Parent  87f258202b0feacb1de72ffc24162b7cbc078163
Add additional calling form to docstring for [xyz]label functions.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-12-06  Rik  <octave@nomad.inbox5.com>
+
+	* plot/xlabel.m, plot/ylabel.m, plot/zlabel.m: Add additional calling
+	form to docstring which returns handle (h)
+
 2010-12-06  Rik  <octave@nomad.inbox5.com>
 
 	* scripts/testfun/speed.m: Overhaul documentation string.
 
 2010-12-03  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/private/__stem__.m: Create a baseline for each stem hggroup.
 	* plot/__go_draw_axes__.m: If not set, then unset the gnuplot
diff --git a/scripts/plot/xlabel.m b/scripts/plot/xlabel.m
--- a/scripts/plot/xlabel.m
+++ b/scripts/plot/xlabel.m
@@ -14,21 +14,23 @@
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {} xlabel (@var{string})
-## @deftypefnx {Function File} {} ylabel (@var{string})
-## @deftypefnx {Function File} {} zlabel (@var{string})
 ## @deftypefnx {Function File} {} xlabel (@var{h}, @var{string})
-## Specify x-, y-, and z-axis labels for the current figure.  If @var{h} is
-## specified then label the axis defined by @var{h}.
+## @deftypefnx {Function File} {@var{h} =} xlabel (@dots{})
+## @deftypefnx {Function File} {} ylabel (@dots{})
+## @deftypefnx {Function File} {} zlabel (@dots{})
+## Specify x-, y-, or z-axis labels for the current axis.  If @var{h} is
+## specified then label the axis defined by @var{h}.  The optional return
+## value @var{h} provides a handle to the created label.
 ## @seealso{plot, semilogx, semilogy, loglog, polar, mesh, contour,
 ## bar, stairs, title}
 ## @end deftypefn
 
 ## Author: jwe
 
 function retval = xlabel (varargin)
 
diff --git a/scripts/plot/ylabel.m b/scripts/plot/ylabel.m
--- a/scripts/plot/ylabel.m
+++ b/scripts/plot/ylabel.m
@@ -15,16 +15,17 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {} ylabel (@var{string})
 ## @deftypefnx {Function File} {} ylabel (@var{h}, @var{string})
+## @deftypefnx {Function File} {@var{h} =} ylabel (@dots{})
 ## @seealso{xlabel}
 ## @end deftypefn
 
 ## Author: jwe
 
 function retval = ylabel (varargin)
 
   [h, varargin, nargin] = __plt_get_axis_arg__ ("ylabel", varargin{:});
diff --git a/scripts/plot/zlabel.m b/scripts/plot/zlabel.m
--- a/scripts/plot/zlabel.m
+++ b/scripts/plot/zlabel.m
@@ -15,16 +15,17 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {} zlabel (@var{string})
 ## @deftypefnx {Function File} {} zlabel (@var{h}, @var{string})
+## @deftypefnx {Function File} {@var{h} =} zlabel (@dots{})
 ## @seealso{xlabel}
 ## @end deftypefn
 
 ## Author: jwe
 
 function retval = zlabel (varargin)
 
   [h, varargin, nargin] = __plt_get_axis_arg__ ("zlabel", varargin{:});
